File: src/main/java/org/railwaystations/rsapi/core/model/Coordinates.java
package org.railwaystations.rsapi.core.model;

import lombok.AllArgsConstructor;
import lombok.Value;

@Value
@AllArgsConstructor
public class Coordinates {

    public static final double ZERO = 0.0;

    double lat;
    double lon;

    public Coordinates() {
        this(ZERO,ZERO);
    }

    public boolean hasZeroCoords() {
        return lat == ZERO && lon == ZERO;
    }

    public boolean isValid() {
        return Math.abs(lat) < 90 && Math.abs(lon) < 180 && !hasZeroCoords();
    }

}

File: src/main/kotlin/org/railwaystations/rsapi/core/model/Coordinates.kt
package org.railwaystations.rsapi.core.model

import kotlin.math.abs

data class Coordinates(
    var lat: Double = 0.0,
    var lon: Double = 0.0,
) {

    constructor() : this(ZERO, ZERO)

    fun hasZeroCoords(): Boolean {
        return lat == ZERO && lon == ZERO
    }

    val isValid: Boolean
        get() = abs(lat) < 90 && abs(lon) < 180 && !hasZeroCoords()

    companion object {
        const val ZERO: Double = 0.0
    }
}
